protobuf-c/protobuf-c.c

```diff
@@ -3161,7 +3161,11 @@ protobuf_c_message_free_unpacked(ProtobufCMessage *message,
 	const ProtobufCMessageDescriptor *desc = message->descriptor;
 	unsigned f;
 
+	if (message == NULL)
```

Comment:
The NULL check needs to move to the top of the function, since there is already a pointer dereference on line 3161:
`const ProtobufCMessageDescriptor *desc = message->descriptor;`